rust/lance/src/index.rs

```diff
@@ -476,6 +536,42 @@ impl DatasetIndexExt for Dataset {
         Ok(())
     }
 
+    async fn describe_indexes<'a, 'b>(
+        &'a self,
+        criteria: Option<IndexCriteria<'b>>,
+    ) -> Result<Vec<Arc<dyn IndexDescription>>> {
```

Comment:
One thing I wanted to do with a new list indexes API is aggregate the index deltas, so we return just one entry per index name. What would you think of doing that here?

Reply:
Yes, I think that is a good idea.  I'll do that.

Reply:
This would change the `IndexDescription` to something like:

```rust
struct IndexDescriptionImpl {
    name: String,
    fragments: Vec<IndexMetadata>,
    details: IndexDetails,
    rows_indexed: u64,
}
```

Reply:
I'm assuming that the `field_ids` and `index_type` will be consistent across all ~~shards~~ segments.